### PR TITLE
rambox: migrate to use AppImage

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Motivation for this change

The motivation comes from PR #108787: AppImage is the official and using it should avoid some issues.

Also, migrating both packages (instead of only rambox-pro like #108787) can make maintaining both packages easier. There is now a `mkRambox` function that abstract most of the build process.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/rambox/default.nix [hunk 1]
```diff
@@ -1,46 +1,23 @@
...
 
 let
-  version = "0.7.7";
-in stdenv.mkDerivation rec {
+  mkRambox = opts: callPackage (import ./rambox.nix opts) { };
+in mkRambox rec {
   pname = "rambox";
-  inherit version;
+
+  version = "0.7.7";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/rambox/default.nix:8]:
```suggestion
  version = "0.7.7";
```

File: pkgs/applications/networking/instant-messengers/rambox/rambox.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    categories = "Network;";
+  });
+
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
+  };
+in appimageTools.wrapType2 rec {
+  inherit name src meta;
+
+  extraPkgs = pkgs: with pkgs; [ xdg_utils ];
```
Comment (thiagokokada) [inline pkgs/applications/networking/instant-messengers/rambox/rambox.nix:23]: Opening links in browser are not working (tested in `rambox`), but I don't know if they worked in `master` or not. The only thing that the old derivation had that seems related is `xdg_utils`, but even adding it here doesn't work it seems.

Some help would be appreciated.

Comment (thiagokokada) [reply pkgs/applications/networking/instant-messengers/rambox/rambox.nix:23]: We could also merge it as is and open an issue to investigate it later.

Comment (thiagokokada) [reply pkgs/applications/networking/instant-messengers/rambox/rambox.nix:23]: Just confirming that it works in `master`.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109020` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>rambox</li>
<li>rambox-pro</li>
</ul>
</details>

File: pkgs/applications/networking/instant-messengers/rambox/rambox.nix (hunk 2 above)
Comment (thiagokokada) [reply pkgs/applications/networking/instant-messengers/rambox/rambox.nix:23]: Just saw this is unnecessary (it is in the default packages from `wrapType2`). Removing it.

File: pkgs/applications/networking/instant-messengers/rambox/default.nix (hunk 1 above)
Comment (thiagokokada) [reply pkgs/applications/networking/instant-messengers/rambox/default.nix]: Fixed in both.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109020` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>rambox-pro</li>
</ul>
</details>

**Comment by thiagokokada** [comment]
@SuperSandro2000 Can you merge this?

**Comment by SuperSandro2000** [comment]
I am not sure if the methodically here is the right one.

**Comment by cawilliamson** [comment]
This gets a huge thumbs up from me (maintainer of the pro variant of rambox.) :+1: 

I think ultimately this makes maintaining the package vastly simpler and less problematic. The pro variant of the package has been broken for several weeks now so it needs to be fixed anyway.

**Review by cawilliamson — APPROVED** [review]
I'm 100% in favour of this change - has my vote! :+1:

**Comment by Ma27** [comment]
I don't have a strong opinion on this, but has anybody checked whether the new package will preserve the settings from my original `rambox`? (This was not the case the last time this got repackaged and if this isn't possible, then :-1: from me).

**Comment by thiagokokada** [comment]
@Ma27 Just tested, seems to work fine (tested with `rambox` and not `rambox-pro`, but no reason to believe this wouldn't work too since AppImage is not a sandbox system like Snaps or Flatpaks).

**Comment by cawilliamson** [comment]
> @Ma27 Just tested, seems to work fine (tested with `rambox` and not `rambox-pro`, but no reason to believe this wouldn't work too since AppImage is not a sandbox system like Snaps or Flatpaks).

The same applies to rambox-pro - no settings loss at all.

**Comment by andresilva** [comment]
Opening URLs seems to be broken after migration to AppImage. Also font and cursor config seem to be ignored.

**Comment by thiagokokada** [comment]
@andresilva Can you reproduce this issue with other AppImage programs? If yes, I think it would be better to try to investigate that issue in `appImageTools`.

Maybe opening an issue?

**Comment by andresilva** [comment]
Yeah the issue seems to be related to AppImage packaging and not specific to Rambox. Will have a look.

**Comment by andresilva** [comment]
https://github.com/NixOS/nixpkgs/issues/110636 I have no clue who to ping though. If someone knows who's responsible for maintaining the AppImage packaging support please cc them on that issue.
